src/GoogleAuthenticator.php

```diff
@@ -152,21 +152,21 @@ public static function getQrCodeUrl($type, $label, $secret, $counter = null, $op
      * This could decode into anything. It's located here as a small helper
      * where code that might need base32 usually also needs something like this.
      *
-     * @param integer $length Exact length of output string
+     * @see https://tools.ietf.org/html/rfc4226#section-4
+     * R6 - The algorithm MUST use a strong shared secret.  The length of
+     * the shared secret MUST be at least 128 bits.  This document
+     * RECOMMENDs a shared secret length of 160 bits.
+     *
+     * @param int $length Exact length of output string
      *
      * @return string Base32 encoded random
      */
     public static function generateRandom($length = 16)
     {
-        $keys = array_merge(range('A','Z'), range(2,7)); // No padding char
-
-        $string = '';
-
-        for ($i = 0; $i < $length; $i++) {
-            $string .= $keys[random_int(0, 31)];
+        if (0 !== $length % 8) {
+            throw new \InvalidArgumentException('length must be divisible by 8');
```

Comment:
Hmm, this will break some peoples code if they have any other value than 16 that is not divisible by 8 at the moment, like 18 was in the test. I was thinking of having a new major version anyways, so I'd propose you remove this from this PR so it can be merged as is.
Then the next major version starts and we can bump versions up and remove/enhance things such as this without having to worry about BC. What do you think?

Reply:
Warning: semi random OTP brain dump below :-)

True. But the problem is if you don't do this you'll get padding if you use "proper" Base32 encoding. 

In my software I don't use the `GoogleAuthenticator` class at all, but just have a 'hardcoded' Base32::encodeUpper(random_bytes(20)). Allowing users to choose is a bug in my opinion, in the next version you could just remove the `$length` and always have 160 bits.

Probably better to ignore this PR then as there is not really a point if you need to accept arbitrary lengths...

I came very close to taking your `Otp` class and just stripping everything I didn't need + hardcoding all parameters as Google Authenticator only supports 6 digits and SHA1 anyway, no need to support anything else, but so far it seems you like most of my PRs, so that's great! :)

Unfortunately, for the time being I need to support PHP 5.4 (CentOS / RHEL 7), but that shouldn't stop you from switching to strict static types for the next major version :-) 

By the way: I tested Google Authenticator, FreeOTP and andOTP with 160 bits secrets and they all work fine.